### ISSUE TITLE
Load DTD from Embedded String Constant

### ISSUE THREAD

**Issue by robloo** [lovettchris/SgmlReader#47]
It used to be possible (version 1.8.21 and below) to load a DTD from a string constant as shown below. This functionality seems to have been removed.

This was done in a library that is UI and resource agnostic. There is no concept of resources or a file system where this code is run and therefore giving it a Uri or Resource to load doesn't make sense here.

```csharp
reader = new SgmlReader()
{
    InputStream        = new StringReader(sgmlFile),
    Dtd                = SgmlDtd.Parse(null, "OFX", new StringReader(OfxData.OfxDtd160), null, null, null),
    DocType            = "OFX",
    CaseFolding        = CaseFolding.ToUpper,
    WhitespaceHandling = WhitespaceHandling.Significant,
};
```

Why was this functionality removed? Can it be added back? It seems supporting loading a DTD directly from a string makes a lot of sense.

```csharp
public class SgmlDtd
{
    public static SgmlDtd Parse(string name, string input)
    {
    }
}
```

**Comment by lovettchris** [comment]
Fixed by https://github.com/lovettchris/SgmlReader/commit/e41635af7e0215e3f4bcbf17437b42ee860a6618

**Comment by robloo** [comment]
Thanks so much for adding this functionality back! I'll take a look at testing it soon and will close this afterwards.

**Comment by lovettchris** [comment]
The fix is included in the new version 1.8.28.

**Comment by robloo** [comment]
Tested and works with 1.8.28. Therefore, I'm closing this. Thanks again!

There was some misunderstanding of the newer SgmlDtd.Parse() signature/usage as well but that was resolved looking at the added tests.
